dvc/remote/base.py

```diff
@@ -162,28 +133,31 @@ def get_file_checksum(self, path_info):
     def _collect_dir(self, path_info):
         dir_info = []
 
-        p_info = copy(path_info)
-        dpath = p_info.path
         for root, dirs, files in self.walk(path_info):
             if len(files) > LARGE_DIR_SIZE:
                 msg = (
                     "Computing md5 for a large directory {}. "
                     "This is only done once."
                 )
-                relpath = self.ospath.relpath(root)
-                logger.info(msg.format(relpath))
-                files = progress(files, name=relpath)
+                title = str(self.path_cls(root))
+                logger.info(msg.format(title))
+                files = progress(files, name=title)
 
             for fname in files:
-                path = self.ospath.join(root, fname)
-                p_info.path = path
-                relpath = self.to_posixpath(self.ospath.relpath(path, dpath))
-
-                checksum = self.get_file_checksum(p_info)
+                file_info = self.path_cls(root) / fname
+                relpath = file_info.relative_to(path_info)
                 dir_info.append(
                     {
-                        self.PARAM_RELPATH: relpath,
-                        self.PARAM_CHECKSUM: checksum,
+                        # NOTE: this is lossy transformation:
+                        #   "hey\there" -> "hey/there"
+                        #   "hey/there" -> "hey/there"
+                        # The latter is fine filename on Windows,
+                        # which will transform to dir/file on back transform.
+                        #
+                        # Yes, this is a BUG, as long as we permit "/" in
+                        # filenames on Windows and "\" on Unix
```

Comment:
/ is not allowed on Windows, right?

how does GIt deal with \ in a path? does not allow them?

Reply:
Let's discuss that in #2059.